ert3/config/_parameters_config.py

```diff
@@ -0,0 +1,179 @@
+import sys
+from typing import Any, Dict, Iterator, List, Optional, Tuple, Union
+from pydantic import (
+    BaseModel,
+    ValidationError,
+    root_validator,
+    validator,
+    StrictInt,
+    StrictStr,
+)
+
+import ert3
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+
+_IndexType = Tuple[Union[StrictStr, StrictInt], ...]
+
+
+class _ParametersConfig(BaseModel):
+    class Config:
+        validate_all = True
+        validate_assignment = True
+        extra = "forbid"
+        allow_mutation = False
+        arbitrary_types_allowed = True
+
+
+def _ensure_valid_name(name: str) -> str:
+    if not name:
+        raise ValueError("Names cannot be of zero length")
+
+    if not all(c.isalpha() or c == "_" for c in name):
+        raise ValueError(
+            "Names are expected to only contain characters and `_`, was: {name}"
+        )
+
+    return name
```

Comment:
I think this is a bit too early to be concerned with what characters are allowed, but I'm fine with it.

Normally I would expect the name to be whatever, but if the name was to be used in URLs or file names, then it would be slugified (https://docs.djangoproject.com/en/dev/glossary/#term-slug) by the application as an implementation detail.

Reply:
I agree! However, there is already a concept of referencing variables from either storage or the workspace in `ensemble.yml` that uses `.` as delimiter. I think the format should be up for discussion, but by not allowing them to be included in the variable names we avoid some odd pitfalls in the imminent future. Slugging is not really an option when the user is referencing these variables later in the configuration file (`storage.my_record`).